scene/2d/canvas_item.cpp

```diff
@@ -820,11 +821,6 @@ float CanvasItem::draw_char(const Ref<Font> &p_font, const Point2 &p_pos, const
 }
 
 void CanvasItem::_notify_transform(CanvasItem *p_node) {
-
```

Comment:
This is a terrible thing to do, it will add a huge overhead to node processing and notification propagation.  Don' t do it, the problem must be somewhere else.

Reply:
I don't understand why we should notify that the transform changed only if the global is invalid, for me they have nothing in common. This global_invalid variable is only here to know whether the global transform is dirty or not, but gets updated only we call the get_global_transform() function. The problem is that, with RemoteTransform2D, we don't call get_global_transform() at all, so the notification is never sent.

On the comment you added you mention "nodes are dirty, will get the notification anyway": this is not true since they get modified only if get_global_transform() was called just before.

Reply:
Seems like the problem is that checking for the global_invalid makes perfect sense when scenes are in a tree structure (which they always are). When something like RemoteTransform comes along, however, that affects something other than its descendants, this model no longer works quite right. 

Would you all think it makes sense to make void _notify_transform(CanvasItem *p_node) virtual and have RemoteTransform override it? Such an approach does seem to go against the grain of the engine (or at least from what I've seen), but RemoteTransform itself also kind of goes against the grain...

Thoughts?

Reply:
I've hit a bug in which an `Area2D` child of a `Sprite` hasn't its transform updated when the latter moves.
Research has taken me here. I think the condition for early return is overoptimizing. I mean, I can understand more or less why `global_invalid` hints, altough I'm with @groud in that such a flag isn't directly related to notifications.

The problem may be with the commented side of the expression (`p_node->xform_change.in_list()`). Shouldn't that be checked as well to avoid discarding non-redundant cases?

Reply:
Just to be clearer, in my bug, if collision shapes are visible, they visually move along with the `Sprite`, but the `Area2D` never updates its transform in the Physics server.

Reply:
After deeper research, I backtrack on my theory.